src/broker/Data.cc

```diff
@@ -345,6 +345,34 @@ struct val_converter {
 
 			return rval;
 			}
+		else if ( type->Tag() == TYPE_LIST )
+			{
+			// lists are just treated as vectors on the broker side.
+			auto lt = type->AsTypeList();
+			auto rval = make_intrusive<ListVal>(TYPE_ANY);
+			auto types = lt->GetTypes();
+			auto pure = lt->IsPure();
+
+			if ( pure )
+				// ListVal doesn't have a type constructor :)
+				rval = make_intrusive<ListVal>(lt->GetPureType()->Tag());
```

Comment:
efficiency nitpick: the previous allocation of `ListVal(TYPE_ANY)` above goes to waste when taking this branch (so would instead either construct using ternary or separate `else` branch, etc.).

Reply:
Yeah, I thought about that too while writing this actually. I just liked the readability of this slightly more - but I can see that that is not worth it :)